### PR TITLE
[TASK] Switch to composer based build

### PR DESCRIPTION
**Pull request by helhum**

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -0,0 +1,63 @@
+{
+    "name": "helhum/typoscript-rendering",
+    "type": "typo3-cms-extension",
+    "description": "Can render a TypoScript path by URL, especially useful for Ajax dispatching",
+    "keywords": [
+        "TYPO3",
+        "rendering",
+        "frontend"
+    ],
+    "homepage": "http://insight.helhum.io/post/104880845705/dont-use-eid-ajax-dispatchers-for-your-extbase",
```
Comment (oliverklee) [inline composer.json]: Should the GitHub issues page also be listed?

Comment (helhum) [inline composer.json]: done

File: composer.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+            "name": "Helmut Hummel",
+            "email": "info@helhum.io",
+            "role": "Developer",
+            "homepage": "http://helhum.io"
+        }
+    ],
+    "license": "GPL-2.0+",
+    "require": {
+        "php": ">=5.3.7",
+        "typo3/cms-core": "^6.2 || ^7.6 || ^8.0"
```
Comment (oliverklee) [inline composer.json]: This extension also works with "inbetween" versions like 7.0, 7.1 etc. I propose to allow this in the dependencies.

File: composer.json [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    },
+    "autoload-dev": {
+        "psr-4": {
+            "Helhum\\TyposcriptRendering\\Tests\\": "Tests",
+            "TYPO3\\CMS\\Core\\Tests\\": ".Build/vendor/typo3/cms/typo3/sysext/core/Tests/"
+        }
+    },
+    "replace": {
+        "typo3_console": "self.version",
+        "typo3-ter/typo3-console": "self.version"
```
Comment (oliverklee) [inline composer.json]: This should be typoscript-rendering/typoscript_rendering, not typo3-console(probably a copy'n'paste error).

File: composer.json (hunk 2 above)
Comment (helhum) [inline composer.json]: That is not really true as some of these versions had some bugs for creating links.
They are also not really relevant any more, os I think this is fine like that

File: ext_emconf.php [hunk 4]
```diff
@@ -11,17 +11,17 @@
...
     array (
       'php' => '5.3.7-7.0.999',
       'typo3' => '6.2.0-8.99.99',
     ),
-    'conflicts' =>
+    'conflicts' => 
     array (
     ),
-    'suggests' =>
+    'suggests' =>
```
Comment (oliverklee) [inline ext_emconf.php]: Are you re-adding the trailing whitespace on purpose here?

File: composer.json (hunk 3 above)
Comment (helhum) [inline composer.json]: true. fixed. thanks

File: ext_emconf.php (hunk 4 above)
Comment (helhum) [inline ext_emconf.php]: yes. I use a script which writes this file with a var_export and this is how PHP generates such an array

File: composer.json [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        "phpunit/phpunit": "~4.8.0"
+    },
+    "autoload": {
+        "psr-4": {
+            "Helhum\\TyposcriptRendering\\": "Classes"
+        }
+    },
+    "autoload-dev": {
+        "psr-4": {
+            "Helhum\\TyposcriptRendering\\Tests\\": "Tests",
```
Comment (oliverklee) [inline composer.json]: With the Tests now being autoloaded, can the hardcoded require be removed?
https://github.com/helhum/typoscript_rendering/blob/master/Tests/Functional/RenderingTest.php#L30

Comment (helhum) [inline composer.json]: Yes! I'm on it already :)

File: composer.json (hunk 3 above)
Comment (helhum) [inline composer.json]: done

**Pull request merged**